src/main/java/org/gridsuite/study/notification/server/NotificationWebSocketHandler.java

```diff
@@ -235,11 +235,19 @@ private void updateConnectionMetrics(WebSocketSession webSocketSession) {
         LOGGER.info("New websocket connection id={} for user={} studyUuid={}, updateType={}", webSocketSession.getId(), userId,
                 webSocketSession.getAttributes().get(FILTER_STUDY_UUID), webSocketSession.getAttributes().get(FILTER_UPDATE_TYPE));
         userConnections.compute(userId, (k, v) -> (v == null) ? 1 : v + 1);
+        updateConnectionMetricsRegistry();
     }
 
     private void updateDisconnectionMetrics(WebSocketSession webSocketSession) {
         var userId = webSocketSession.getHandshakeInfo().getHeaders().getFirst(HEADER_USER_ID);
         LOGGER.info("Websocket disconnection id={} for user={}", webSocketSession.getId(), userId);
         userConnections.computeIfPresent(userId, (k, v) -> v > 1 ? v - 1 : null);
+        updateConnectionMetricsRegistry();
+    }
+
+    private void updateConnectionMetricsRegistry() {
+        multiGauge.register(userConnections.entrySet().stream()
```

Comment:
Perhaps add here the builder ?
`MultiGauge.builder(USERS_METER_NAME).register(meterRegistry)`